### DISCUSSION TITLE
Primer Planteo Del Proyecto

### DISCUSSION THREAD

**Discussion by AstorArico72** [AstorArico72/Proyecto-Transversal-1#2] [General]
### Mi idea para la tabla SQL es ésta (revisada 5/12 10:46 AM):
**Alumnos**

INT (6) ID_Alumno _--ó "legajo"_

> La clave primaria para ésta tabla. No nulo, preferiblemente no autoincremental y único.

> También podría ser autoincremental, pero no es recomendable en mi opinión.

> Por ejemplo "103392"

VARCHAR (50) Nombre_Alumno

> No nulo, pero no único.

> Por ejemplo "John Doe"

BOOL Aprobado

_Como ejemplo de toda una fila: 103392, "John Doe", False._

**Materias**

VARCHAR (50) Nombre_Materia
INT (3) ID_Materia

> Llave primaria para ésta tabla. No nulo, no autoincremental.

_Como ejemplo de una fila: 101, "Matemáticas I"_

**Inscripción**
_Ésta tabla no tiene llave primaria._
INT (6) ID_Alumno

> Llave foránea, enlace a Alumno.ID_Alumno

VARCHAR (50) Nombre_Alumno

> Llave foránea, enlace a Alumno.Nombre_Alumno

INT (3) ID_Materia

> Llave foránea, enlace a Materias.ID_Materia.

VARCHAR (50) Nombre_Materia

> Llave foránea, enlace a Materias.Nombre_Materia.

BOOL Aprobado
FLOAT (1) Nota

_Por ejemplo: 103392, "John Doe", 101, "Matemáticas I", False, 5.91_

### Se agradece cualquier contradicción o remarque. Corregí algunos errores horribles.

**Comment by AstorArico72** [comment]
En adición a éso, propongo poner la base de datos en una nueva rama "SQL_DB".

**Comment by AstorArico72** [comment]
Además, si vamos a enviar archivos comprimidos, convengamos en usar el formato ".zip" porque no puedo abrir archivos ".rar" en mi ordenador Linux.
Y los ordenadores Windows no pueden abrir archivos ".tar" (y por extensión ".tar.gz").

**Reply by PabloBarrionuevo** [reply]
Todo es posible, usemos .zip entonces.

**Comment by PabloBarrionuevo** [comment]
Perdón no he estado asistiendo a clases, vi el último video, propongo para usar en mysql local, si está bien comparto el archivo
[LAB1_Grupo_1_TransversalUniversidad.pdf](https://github.com/AstorArico72/Proyecto-Transversal-1/files/6446531/LAB1_Grupo_1_TransversalUniversidad.pdf)

**Comment by PabloBarrionuevo** [comment]
El tema de aprobado no debería estar en alumnos (por Universidad.sql), bajo mi punto de vista, sería algo propio de una cursada o inscripción a una materia de la universidad, el cual no tiene nada que ver con otras cosas como la materia por ejemplo: si el alumno aprobó la materia, para mi, eso no debe ir ni en materia ni en alumno, para eso tendríamos la tabla de relaciones entre alumnos y materias, donde colocar cosas como la nota, si aprobó, u otros como comentarios, etc.
Si no quedaría como dato de un alumno:
Jhon Doe, esta ¿aprobado = true | false?, y esto qué significa para la universidad.
En lugar de aprobado podríamos usarlo como si el alumno esta activo en la universidad (si no ha claudicado, o se ha dado de baja):
Jhon Doe, esta ¿activo = true | false?, tiene mas sentido.
Agregar comentarios a la tabla de alumnos es buena idea.
